pkg/controller/volume/persistentvolume/binder_test.go

```diff
@@ -517,6 +519,114 @@ func TestSync(t *testing.T) {
 			newClaimArray("claim13-5", "uid13-5", "1Gi", "volume13-5", v1.ClaimBound, nil, annBoundByController, annBindCompleted),
 			noevents, noerrors, testSyncClaim,
 		},
+
+		// PVC with VolumeMode
+		{
+			// syncVolume binds a requested block claim to a block volume
+			"14-1 - binding to volumeMode block",
+			withVolumeVolumeMode(&modeBlock, newVolumeArray("volume14-1", "10Gi", "", "", v1.VolumePending, v1.PersistentVolumeReclaimRetain, classEmpty)),
+			withVolumeVolumeMode(&modeBlock, newVolumeArray("volume14-1", "10Gi", "uid14-1", "claim14-1", v1.VolumeBound, v1.PersistentVolumeReclaimRetain, classEmpty, annBoundByController)),
+			withClaimVolumeMode(&modeBlock, newClaimArray("claim14-1", "uid14-1", "1Gi", "", v1.ClaimPending, nil)),
+			withClaimVolumeMode(&modeBlock, newClaimArray("claim14-1", "uid14-1", "1Gi", "volume14-1", v1.ClaimBound, nil)),
```

Comment:
missing `annBoundByController` and `annBindCompleted`, ie.:
```
withClaimVolumeMode(&modeBlock, newClaimArray("claim14-1", "uid14-1", "1Gi", "volume14-1", v1.ClaimBound, nil, annBoundByController, annBindCompleted)),
```
 
And it's missing everywhere you expect a claim to be bound in the end.

Reply:
and claim size should be "10Gi" everywhere (unless you modify it with `withCapacity()`), the test framework is quite stupid here and expects the PV and PVC have the same size in most of the tests.

Reply:
ack - I will polish up these tests and change the capacity, double check names, etc...